src/components/presentational/LineChart/LineChart.tsx

```diff
@@ -0,0 +1,139 @@
+import { format } from 'date-fns'
+import { LayoutContext } from '..'
+import { CartesianGrid, Line, LineChart as ReChartsLineChart, ReferenceLine, ResponsiveContainer, Tooltip, XAxis, YAxis } from 'recharts'
+import "./LineChart.css"
+import { ColorDefinition, resolveColor } from '../../../helpers/colors'
+import { useContext } from 'react'
+import React from 'react'
+
+export type LineChartPreviewState = "Default";
+
+export interface LineChartThreshold {
+    value: number
+    referenceLineColor?: ColorDefinition
+    overThresholdLineColor?: ColorDefinition
+}
+
+export interface LineChartDataPoint {
+    value: number,
+    date: Date
+}
+
+export interface LineChartAxis {
+    yRange?: number[]
+    yIncrement: number;
+    xRange?: number[];
+    xIncrement: number;
+}
+
+export interface LineChartProps {
+    data: LineChartDataPoint[],
+    axis?: LineChartAxis,
+    lineColor?: ColorDefinition,
+    thresholds?: LineChartThreshold[],
+    innerRef?: React.Ref<HTMLDivElement>
+}
+
+export default function MdhLineChart(props: LineChartProps) {
+    const thresholdKey = "_threshold";
+    const data: LineChartDataPoint[] = props.data;
+    let layoutContext = useContext(LayoutContext);
+    const defaultLineColor = resolveColor(layoutContext.colorScheme, props.lineColor) || '#bbb';
+
+    var axis: LineChartAxis = props.axis ?? { yRange: [0, 100], yIncrement: 5, xRange: [0, 100], xIncrement: 1 }
+    if (!props.axis || (props.axis && !props.axis.yRange)) {
+        const yValues: number[] = data.map(d => d.value);
+        const yMax = Math.max(...yValues);
+        axis.yRange = [0, yMax];
```

Comment:
Curious why you need to do this...I think just not supplying a domain will be sufficient